src/frontend/src/instance/log_handler.rs

```diff
@@ -24,15 +24,15 @@ use servers::error::{
     AuthSnafu, Error as ServerError, ExecuteGrpcRequestSnafu, PipelineSnafu, Result as ServerResult,
 };
 use servers::interceptor::{LogIngestInterceptor, LogIngestInterceptorRef};
-use servers::query_handler::LogHandler;
+use servers::query_handler::PipelineHandler;
 use session::context::QueryContextRef;
 use snafu::ResultExt;
 
 use crate::instance::Instance;
 
 #[async_trait]
-impl LogHandler for Instance {
-    async fn insert_logs(
+impl PipelineHandler for Instance {
+    async fn transform(
```

Comment:
It's not actually "transform". At this point, the data have been processed and divided into separate columns if possible. It's more like purely inserting.

Reply:
I see. The execution is done in `exec_mut` of the pipeline. Before calling `insert`